File: personal/fallout76-cheat/Fallout76Cheat/routine.cpp
#include "stdafx.h"

#include "routine.h"

namespace cheat {

	Routine::Routine(bool disabled_, std::string cfg_filename, std::string name_, std::string desc_) : FeatureBase(disabled_, cfg_filename, name_, desc_), enabled(false) {
		get_config()->SetEntry("start_enabled", enabled);
	}
	Routine::Routine() : FeatureBase(false, "internal_routine.cfg", "", ""), enabled(false) {}
	Routine::~Routine() {
		set_enabled(false);
	}
	bool Routine::is_enabled() const {
		return enabled;
	}
	void Routine::Toggle() {
		set_enabled(!is_enabled());
	}
	void Routine::set_enabled(bool enabled_) {
		bool prev_enabled = is_enabled();
		enabled = enabled_;
		if (prev_enabled != is_enabled()) {
			OnToggled();
		}
	}

}  // namespace cheat

File: personal/pklmao/src/main/java/org/bitbucket/pklmao/routine/Routine.java
package org.bitbucket.pklmao.routine;

import javax.swing.AbstractButton;

import org.bitbucket.pklmao.cfg.Configuration;

public interface Routine {
	String id();
	String name();
	String desc();
	void setEnabled(boolean enabled);
	boolean isEnabled();
	Configuration config();
	AbstractButton toggleBtn();
}
